src/main/java/com/example/schooloperationsystem/mapper/PupilMapper.java

```diff
@@ -0,0 +1,26 @@
+package com.example.schooloperationsystem.mapper;
+import com.example.schooloperationsystem.entity.Pupil;
+import com.example.schooloperationsystem.rest.dto.response.PupilDetailsDto;
+import org.springframework.stereotype.Component;
+import java.util.ArrayList;
+import java.util.List;
+/**
+ * @author Gurgen Bayburdyan
+ */
+@Component
+public class PupilMapper {
+    public PupilDetailsDto map(Pupil pupil) {
+        PupilDetailsDto detailsDto = new PupilDetailsDto();
```

Comment:
please add logs such as

log.trace("Mapping pupil - {} to pupil details dto", pupil);